modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/TWKBAttributeIO.java

```diff
@@ -0,0 +1,231 @@
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+import org.geotools.data.Base64;
+import org.geotools.data.DataSourceException;
+import org.geotools.geometry.jts.LiteCoordinateSequence;
+import org.geotools.geometry.jts.LiteCoordinateSequenceFactory;
+import org.locationtech.jts.geom.CoordinateSequence;
+import org.locationtech.jts.geom.CoordinateSequences;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.LinearRing;
+import org.locationtech.jts.geom.MultiLineString;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.Polygon;
+import org.locationtech.jts.io.ByteArrayInStream;
+import org.locationtech.jts.io.WKBWriter;
+
+public class TWKBAttributeIO {
+    TWKBReader wkbr;
+    ByteArrayInStream inStream = new ByteArrayInStream(new byte[0]);
+    GeometryFactory gf;
+
+    /**
+     * A variation of TWKBReader tha optimizes conversion of collapsed point expressed using
+     * LiteCoordinateSequece
+     */
+    static final class LiteTWKBReader extends TWKBReader {
+
+        public LiteTWKBReader(GeometryFactory geometryFactory) {
+            super(geometryFactory);
+        }
+
+        protected CoordinateSequence readCoordinateSequence(int numPts, TWKBMetadata metadata)
+                throws IOException {
+            if (!(csFactory instanceof LiteCoordinateSequenceFactory)) {
+                return super.readCoordinateSequence(numPts, metadata);
+            }
+
+            int dims = metadata.getDims();
+            LiteCoordinateSequence seq = (LiteCoordinateSequence) csFactory.create(numPts, dims);
+            double[] ordinates = new double[numPts * dims];
+            int k = 0;
+            for (int i = 0; i < numPts; i++) {
+                for (int j = 0; j < dims; j++) {
+                    double ordinateDelta = readNextDouble(metadata.getScale(j));
+                    double value = metadata.valueArray[j] + ordinateDelta;
+                    metadata.valueArray[j] = value;
+                    ordinates[k++] = value;
+                }
+            }
+            seq.setArray(ordinates);
+
+            return seq;
+        }
+    }
+
+    public TWKBAttributeIO() {
+        this(new GeometryFactory());
+    }
+
+    public TWKBAttributeIO(GeometryFactory gf) {
+        wkbr = new LiteTWKBReader(gf);
+        this.gf = gf;
+    }
+
+    public void setGeometryFactory(GeometryFactory gf) {
+        if (gf != this.gf) {
+            this.gf = gf;
+            wkbr = new LiteTWKBReader(gf);
+        }
+    }
+
+    public GeometryFactory getGeometryFactory() {
+        return gf;
+    }
+
+    /**
+     * This method will convert a Well Known Binary representation to a JTS Geometry object.
+     *
+     * @param wkb te wkb encoded byte array
```

Comment:
wkb te wkb?  

Can you clarify this and update the WTB references below in this comment block?